app/controllers/CyaCreateRecordController.scala

```diff
@@ -63,9 +68,31 @@ class CyaCreateRecordController @Inject() (
   }
 
   def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    auditService
-      .auditFinishCreateGoodsRecord(request.eori, request.affinityGroup, request.userAnswers)
-      .map(_ => Redirect(routes.CreateRecordSuccessController.onPageLoad().url))
+    CreateGoodsRecordRequest.build(request.userAnswers, request.eori) match {
+      case Right(model) =>
+        goodsRecordConnector
+          .submitGoodsRecordUrl(model, request.eori)
+          .flatMap { goodsRecordResponse =>
+            auditService
+              .auditFinishCreateGoodsRecord(request.eori, request.affinityGroup, request.userAnswers)
+              .map { _ =>
+                cleanseUserAnswers(request.userAnswers)
+                Redirect(routes.CreateRecordSuccessController.onPageLoad(goodsRecordResponse.recordId))
+              }
+          }
+      case Left(errors) => Future.successful(logErrorsAndContinue(errors))
+    }
+  }
+
+  def cleanseUserAnswers(userAnswers: UserAnswers): Done = {
```

Comment:
can't get this to work - helpful answers welcome

Reply:
Why are we returning Done?

Reply:
because i've no idea what i am doing >_<

Reply:
removed and will implement in another ticket